Formula/iodine.rb

```diff
@@ -0,0 +1,21 @@
+class Iodine < Formula
+  desc "Tunnel IPv4 traffic through a DNS server"
+  homepage "https://code.kryo.se/iodine"
+  url "https://github.com/yarrick/iodine/archive/refs/tags/v0.7.0.tar.gz"
+  sha256 "7281e2301804e48029877bab27134f7c0eb04567da4d21a6fcbaa7265cb5849e"
+  license "ISC"
+  head "https://github.com/yarrick/iodine.git", branch: "master"
+  depends_on "cmake" => :build
+
+  def install
+    system "make"
+    bin.install "bin/iodine"
+    bin.install "bin/iodined"
+    man8.install "man/iodine.8"
+  end
+
+  test do
+    system "#{bin}/iodine", "-h"
+    system "#{bin}/iodined", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Unfortunately all switches in the code path with the exception of the version character require root privileges for either of the two binaries:

chrishellberg@chriss-MacBook-Pro src % iodined -u chrishellberg -b 10000
iodined: Run as root and you'll be happy.
chrishellberg@chriss-MacBook-Pro src %

chrishellberg@chriss-MacBook-Pro src % iodine -u chrishellberg
iodine: Run as root and you'll be happy.
chrishellberg@chriss-MacBook-Pro src %

There are already series of tests of the binary using libcheck as part of the make process:

chrishellberg@chriss-MacBook-Pro iodine-0.7.0 % make test
OS is DARWIN, arch is arm64
!! The check library is required for compiling and running the tests
!! Get it at http://check.sf.net
Running suite(s): iodine
100%: Checks: 68, Failures: 0, Errors: 0
chrishellberg@chriss-MacBook-Pro iodine-0.7.0 %

But this was disallowed as per latest brew Foruma audit checking (no run-time checks) when I tried to include:

system "make", "test"

Reply:
So we can run a command and check for `iodined: Run as root and you'll be happy.` as output, sounds good.